huff0/_generate/gen.go

```diff
@@ -0,0 +1,366 @@
+package main
+
+//go:generate go run gen.go -out decompress_amd64.s -stubs delme.go -pkg=huff0
+
+import (
+	"flag"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	_ "github.com/klauspost/compress"
+
+	. "github.com/mmcloughlin/avo/build"
+	"github.com/mmcloughlin/avo/buildtags"
+	"github.com/mmcloughlin/avo/gotypes"
+	. "github.com/mmcloughlin/avo/operand"
+	"github.com/mmcloughlin/avo/reg"
+)
+
+func main() {
+	flag.Parse()
+	out := flag.Lookup("out")
+	os.Remove(filepath.Join("..", out.Value.String()))
+	stub := flag.Lookup("stubs")
+	if stub.Value.String() != "" {
+		os.Remove(stub.Value.String())
+		defer os.Remove(stub.Value.String())
+	}
```

Comment:
What is going on here?

The `-stubs` flag is not required. If you don't want stubs, just don't provide the flag.

What's the reason for the `-out` lookup?

Reply:
This is mostly leftovers from when we had to rewrite to get `#ifdefs` working, and it left 0 byte Go files when it failed, which would make it impossible to regenerate.

I will remove it in #569

Reply:
Ah I believe you are running into this: https://github.com/mmcloughlin/avo/issues/57